src/overlays/actors/ovl_En_Fg/z_en_fg.c

```diff
@@ -21,40 +47,418 @@ const ActorInit En_Fg_InitVars = {
+                this->actionFunc = func_80A2D778;
+            }
+    }
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D4B8.asm")
+void func_80A2D778(EnFg* this, GlobalContext* globalCtx) {
+    Actor* ac;
+    s32 pad;
+    s16 rotY;
+    s16 rotX;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D778.asm")
+    switch (func_80A2D42C(this)) {
+        case FG_DMGEFFECT_ARROW:
+            this->actor.flags &= ~1;
+            this->skelAnime.animPlaybackSpeed = 0.0f;
+            ac = this->collider.base.ac;
+            rotY = ac->world.rot.y;
+            rotX = ac->world.rot.x;
+            this->actor.scale.x = fabsf(Math_CosS(rotY) * 0.01f);
+            this->actor.scale.y = fabsf(Math_CosS(rotX) * 0.01f);
+            this->actor.scale.z = fabsf(Math_SinS(rotY) * 0.01f);
+            this->actor.scale.x = CLAMP_MIN(this->actor.scale.x, 0.001f);
+            this->actor.scale.y = CLAMP_MIN(this->actor.scale.y, 0.001f);
+            this->actor.scale.z = CLAMP_MIN(this->actor.scale.z, 0.001f);
+            this->actionFunc = EnFg_DoNothing;
+            break;
+        case FG_DMGEFFECT_HOOKSHOT:
+            break;
+        case FG_DMGEFFECT_EXPLOSION:
+            this->actor.flags &= ~1;
+            Audio_PlayActorSound2(this, 0x28E3);
+            func_80A2D280(&this->skelAnime, 0);
+            this->actor.params = 5;
+            this->skelAnime.animPlaybackSpeed = 0.0f;
+            ac = this->collider.base.ac;
+            this->actor.world.rot.y = Math_Vec3f_Yaw(&ac->world.pos, &this->actor.world.pos);
+            this->actor.shape.rot = this->actor.world.rot;
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 3.0f;
+            this->actor.gravity = -0.8f;
+            this->unk_2FA = 1;
+            this->timer = 0;
+            this->actionFunc = func_80A2D9DC;
+            break;
+        default:
+            if (func_801378B8(&this->skelAnime, 8.0f)) {
+                this->skelAnime.animCurrentFrame = 8.0f;
+                this->skelAnime.animPlaybackSpeed = 0.0f;
+            }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D9CC.asm")
+            if ((this->actor.velocity.y <= 0.0f) && (this->actor.bgCheckFlags & 1)) {
+                func_80A2D280(&this->skelAnime, 0);
+                this->actionFunc = func_80A2D4B8;
+                this->actor.velocity.y = 0.0f;
+            } else {
+                Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+            }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D9DC.asm")
+void EnFg_DoNothing(EnFg* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/EnFg_Init.asm")
+void func_80A2D9DC(EnFg* this, GlobalContext* globalCtx) {
```

Comment:
I'm thinking this is `EnFg_Jump`

Reply:
Based on how the frog reactions when hit by the explosion (see video below), I'm thinking more `EnFg_Knockback`. Jump makes me think of the jumping animation when idle.

https://discord.com/channels/688807550715560050/688852030403379277/844663571522125834

Reply:
Works for me. You know the system better than I do.